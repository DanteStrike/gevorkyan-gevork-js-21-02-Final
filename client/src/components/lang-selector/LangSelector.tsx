import React from 'react';
import {Select} from 'antd';
import useAppTranslation from '../../hooks/use-app-translation';

function LangSelector() {
  const {i18n} = useAppTranslation();

  const handleLangChange = (lang: string) => {
    i18n.changeLanguage(lang);
  };

  return (
    <Select defaultValue={i18n.resolvedLanguage} value={i18n.resolvedLanguage} onChange={handleLangChange}>
      <Select.Option value="ru">ru-RU</Select.Option>
      <Select.Option value="en">en-EN</Select.Option>
    </Select>
  );
}

export default LangSelector;
