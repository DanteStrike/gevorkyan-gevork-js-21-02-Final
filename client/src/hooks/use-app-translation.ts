import {useTranslation} from 'react-i18next';

const useAppTranslation = (nestSpace?: string) => useTranslation(`translation`, {keyPrefix: nestSpace});

export default useAppTranslation;
