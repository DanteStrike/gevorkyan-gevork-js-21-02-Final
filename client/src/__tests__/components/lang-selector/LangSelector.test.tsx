import {shallow} from 'enzyme';
import React from 'react';
import * as i18next from '../../../hooks/use-app-translation';
import LangSelector from '../../../components/lang-selector/LangSelector';

describe(`Component LangSelector should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<LangSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe(`Component LangSelector should work correctly`, () => {
  it(`change test`, () => {
    const spy = {
      i18n: {
        changeLanguage: jest.fn(() => Promise.resolve(() => {})),
        resolvedLanguage: `ru`,
      },
    };
    jest.spyOn(i18next, 'default').mockReturnValue(spy as any);

    const wrapper = shallow(<LangSelector />);
    expect(spy.i18n.changeLanguage).toHaveBeenCalledTimes(0);

    wrapper.simulate(`change`, `any`);
    expect(spy.i18n.changeLanguage).toHaveBeenCalledTimes(1);
    expect(spy.i18n.changeLanguage).toHaveBeenLastCalledWith(`any`);
  });
});
