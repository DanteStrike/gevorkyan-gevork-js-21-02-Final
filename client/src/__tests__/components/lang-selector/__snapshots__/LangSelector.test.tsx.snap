// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LangSelector should render correctly snapshot default test 1`] = `
<ForwardRef(InternalSelect)
  onChange={[Function]}
>
  <Option
    value="ru"
  >
    ru-RU
  </Option>
  <Option
    value="en"
  >
    en-EN
  </Option>
</ForwardRef(InternalSelect)>
`;
